components/scream/src/physics/p3/p3_functions.hpp

```diff
@@ -244,9 +262,16 @@ struct Functions
     view_2d_table& vn_table_vals, view_2d_table& vm_table_vals, view_2d_table& revap_table_vals,
     view_1d_table& mu_r_table_vals, view_dnu_table& dnu);
 
+  static void init_kokkos_tables_new(
+    const uview_2d<Scalar>& vn_table_vals, const uview_2d<Scalar>& vm_table_vals, const uview_2d<Scalar>& revap_table_vals,
+    const uview_1d<Scalar>& mu_r_table_vals, const uview_1d<Scalar>& dnu);
+
   static void init_kokkos_ice_lookup_tables(
     view_ice_table& ice_table_vals, view_collect_table& collect_table_vals);
 
+  static void init_kokkos_ice_lookup_tables_new(
+    const uview_ND<Scalar,4>& ice_table_vals, const uview_ND<Scalar,5>& collect_table_vals);
+
```

Comment:
@bartgol This obviously is not the final form, but I am having an issue that the original functions (`init_kokkos_ice_lookup_tables()` and `init_kokkos_tables()`) cannot accept unmanaged views, but the memory buffer in `atmosphere_microphysics` cannot define these `view_*_table` which are type `view<Scalar[x][y]...[z]>`.

1. Is that correct? Am I wrong to assume buffer views need to be unmanaged?
2. I don't think we want two versions of each `init_tables()`. Should `atmosphere_microphysics` just define its own version locally? Should we change all impl to the `init_tables(unmanaged_view_type....)`?

Reply:
I don't think you want to use the atm memory buffer. You need these tables to not be overwritten by others, right? Why not just declaring them as local vars to `P3Microphysics`? Or maybe in `P3Functions::Infrastructure`?

Reply:
Of course.. Ok, I'll make them local..

Reply:
Nvm, just read the comment below about `P3LookupTables`.

So now I'm confused. Why do you need unmanaged views?